examples/mcmc.jl

```diff
@@ -137,26 +127,27 @@ function data_generating_procedure(t::Int64)
         _choices = Gen.get_submap(choices, addr)
         Gen.set_submap!(obs, addr, _choices)
     end
-
-    #=  plt = lineplot(1:t, get_zs(trace),
-                   title="Height of ball", name="observation with res $(trace[:prior => 1 => :restitution])",
-                   xlabel="t", ylabel="z", canvas=DotCanvas,
-                   border=:ascii) =#
-    plt = plot(1:t, get_zs(trace),title="Height of ball", xlabel="t", ylabel="z",label="Observation")
     
-    return (gargs, obs, plt)
+    return (gargs, obs)
 
 end
 
 function main()
 
     t = 60 # 1 second of observations
-    (gargs, obs, plt) = data_generating_procedure(t)
+    (gargs, obs) = data_generating_procedure(t)
+
+    zs = [get_submap(obs, :kernel => i => :observe => 1)[:position][3] for i in 1:t]
+    plt = plot(1:t, zs, title="Height of ball", xlabel="t", ylabel="z", label="Observation")
     display(plt)
     
-    update_vis(trace) = display(plot!(plt, 1:t, get_zs(trace), label="res $(trace[:prior => 1 => :restitution])"))
-    (tr, aratio) = inference_procedure(gargs, obs, update_vis)
-    print(tr[:prior => 1])
+    update_vis(trace) = display(plot!(plt, 1:t, get_zs(trace), label="res $(round(trace[:prior => 1 => :restitution]; digits=3))"))
+    (tr, aratio, mass_log, res_log, scores) = inference_procedure(gargs, obs, update_vis)
```

Comment:
this will sound pedantic, but it will come up later when you practice with more complicated inference procedures; please remove `update_viz` from `inference_procedure`. you have several options: 
- return a vector of traces that you can then extract latents, log scores, positions etc
-  in addition to the vectors you currently return, you could also return the z positions of each trace (e.g., a nested vector of vectors, you could also used a matrix).

Reply:
i see. is there any other way of live monitoring during the inference procedure?

Reply:
what you did is legal, i just prefer minimal functions, especially for these examples. although in this case, I'm not sure what would be missing by storing the intermediate results in an array?

Reply:
on my computer the inference runs for two or three minutes already, this is why i liked it to see live progress and understand better what is going on over time. Of course, I can also animate this post-hoc to visualize the inference process step-wise.

Reply:
i can understand that. let's move on - i may make further refinements to this later